src/embeddings/tools/sgns.cpp

```diff
@@ -0,0 +1,573 @@
+/**
+ * @file sgns.cpp
+ * @author Shawn LeMaster
+ * @author Matt Kelly
+ *
+ * Builds word embedding vectors using the Skip-Gram Negative Sampling method.
+ *
+ * @see https://code.google.com/archive/p/word2vec/
```

Comment:
I've added something but truly have no idea if I've done it in a good way or not...